akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -576,8 +613,14 @@ final class ByteStringBuilder extends Builder[Byte, ByteString] {
         _length += bs.length
       case xs: WrappedArray.ofByte ⇒
         putByteArrayUnsafe(xs.array.clone)
+      case seq: collection.IndexedSeq[Byte] if shouldResizeTempFor(seq.length) ⇒
+        //we will have to resize and copy anyways, might as well flush it
+        clearTemp()
+        _builder += ByteString.ByteString1(seq)
+        _length += seq.length
```

Comment:
ah, so the main reason this is here is to avoid resizing the temp buffer if the next IndexedSeq will flood it out. (causing another copy and allocation just for the temp buffer). The temp buffer itself is also copied again, so if we can produce a compact array from it right off the bat, we avoid 1 definite copy, and 1 potential copy for the temp buffer resize. This also cleans up the builder nicely where the temp buffer isnt even used or allocated unless you're feeding it individual Bytes or some TraversableOnce.

This should probably be for any Seq (not just indexedseq) as it should happen with anything we know the length of (TraversableOnce etc. should always go to temp as we dont have a size hint). 

The only downside I see of doing this is that we will end up producing more sparse ByteStrings instead of dense ones when constructing from IndexedSeqs. So the tradeoff ends up being: less copies, but more sparseness (and sparseness may come in handy with TcpConnection and ByteString#headByteString.canWrapAsByteBuffer, as we can avoid another copy there and potentially dereference more buffers over time).